### PR TITLE
feat(object): added object resource

### PR DESCRIPTION
**Pull request by Codelax**

### CODE REVIEW

File: scaleway/resource_object_test.go [hunk 1]
```diff
@@ -0,0 +1,463 @@
...
+
+	"github.com/aws/aws-sdk-go/aws/awserr"
+	"github.com/aws/aws-sdk-go/service/s3"
+	sdkacctest "github.com/hashicorp/terraform-plugin-sdk/v2/helper/acctest"
+	"github.com/hashicorp/terraform-plugin-sdk/v2/helper/resource"
+	"github.com/hashicorp/terraform-plugin-sdk/v2/terraform"
+	"github.com/scaleway/scaleway-sdk-go/scw"
+)
+
+func TestAccScalewayObject_Basic(t *testing.T) {
```
Comment (Monitob) [inline scaleway/resource_object_test.go:15]: Can we add a test for the hash update

**Review by Monitob — CHANGES_REQUESTED** [review]

File: scaleway/resource_container.go [hunk 2]
```diff
@@ -288,7 +288,7 @@ func resourceScalewayContainerUpdate(ctx context.Context, d *schema.ResourceData
 
 	if d.HasChanges("environment_variables") {
 		envVariablesRaw := d.Get("environment_variables")
-		req.EnvironmentVariables = expandMapStringStringPtr(envVariablesRaw)
+		req.EnvironmentVariables = expandMapPtrStringString(envVariablesRaw)
```
Comment (remyleone) [inline scaleway/resource_container.go:291]: This should not be in this PR

File: scaleway/helpers_container.go [hunk 3]
```diff
@@ -59,7 +59,7 @@ func setCreateContainerRequest(d *schema.ResourceData, region scw.Region) (*cont
 
 	// optional
 	if envVariablesRaw, ok := d.GetOk("environment_variables"); ok {
-		req.EnvironmentVariables = expandMapStringStringPtr(envVariablesRaw)
+		req.EnvironmentVariables = expandMapPtrStringString(envVariablesRaw)
```
Comment (remyleone) [inline scaleway/helpers_container.go:62]: This should not be in this PR

File: scaleway/resource_function_namespace.go [hunk 4]
```diff
@@ -146,7 +146,7 @@ func resourceScalewayFunctionNamespaceUpdate(ctx context.Context, d *schema.Reso
 	}
 
 	if d.HasChanges("environment_variables") {
-		req.EnvironmentVariables = expandMapStringStringPtr(d.Get("environment_variables"))
+		req.EnvironmentVariables = expandMapPtrStringString(d.Get("environment_variables"))
```
Comment (remyleone) [inline scaleway/resource_function_namespace.go:149]: This should not be in this PR

File: scaleway/resource_function_namespace.go [hunk 5]
```diff
@@ -75,7 +75,7 @@ func resourceScalewayFunctionNamespaceCreate(ctx context.Context, d *schema.Reso
 
 	ns, err := api.CreateNamespace(&function.CreateNamespaceRequest{
 		Description:          expandStringPtr(d.Get("description").(string)),
-		EnvironmentVariables: expandMapStringStringPtr(d.Get("environment_variables")),
+		EnvironmentVariables: expandMapPtrStringString(d.Get("environment_variables")),
```
Comment (remyleone) [inline scaleway/resource_function_namespace.go:78]: This should not be in this PR

File: scaleway/resource_function.go [hunk 6]
```diff
@@ -294,7 +294,7 @@ func resourceScalewayFunctionUpdate(ctx context.Context, d *schema.ResourceData,
 	updated := false
 
 	if d.HasChange("environment_variables") {
-		req.EnvironmentVariables = expandMapStringStringPtr(d.Get("environment_variables"))
+		req.EnvironmentVariables = expandMapPtrStringString(d.Get("environment_variables"))
```
Comment (remyleone) [inline scaleway/resource_function.go:297]: This should not be in this PR

File: scaleway/resource_function.go [hunk 7]
```diff
@@ -150,7 +150,7 @@ func resourceScalewayFunctionCreate(ctx context.Context, d *schema.ResourceData,
 
 	req := &function.CreateFunctionRequest{
 		Description:          expandStringPtr(d.Get("description").(string)),
-		EnvironmentVariables: expandMapStringStringPtr(d.Get("environment_variables")),
+		EnvironmentVariables: expandMapPtrStringString(d.Get("environment_variables")),
```
Comment (remyleone) [inline scaleway/resource_function.go:153]: This should not be in this PR

File: scaleway/resource_container_namespace.go [hunk 8]
```diff
@@ -149,7 +149,7 @@ func resourceScalewayContainerNamespaceUpdate(ctx context.Context, d *schema.Res
 	}
 
 	if d.HasChanges("environment_variables") {
-		req.EnvironmentVariables = expandMapStringStringPtr(d.Get("environment_variables"))
+		req.EnvironmentVariables = expandMapPtrStringString(d.Get("environment_variables"))
```
Comment (remyleone) [inline scaleway/resource_container_namespace.go:152]: This should not be in this PR

File: scaleway/resource_container_namespace.go [hunk 9]
```diff
@@ -82,7 +82,7 @@ func resourceScalewayContainerNamespaceCreate(ctx context.Context, d *schema.Res
 
 	ns, err := api.CreateNamespace(&container.CreateNamespaceRequest{
 		Description:          expandStringPtr(d.Get("description").(string)),
-		EnvironmentVariables: expandMapStringStringPtr(d.Get("environment_variables")),
+		EnvironmentVariables: expandMapPtrStringString(d.Get("environment_variables")),
```
Comment (remyleone) [inline scaleway/resource_container_namespace.go:85]: This should not be in this PR

File: scaleway/helpers.go [hunk 10]
```diff
@@ -151,6 +151,17 @@ func parseRegionalID(regionalID string) (region scw.Region, id string, err error
...
 
+// parseRegionalNestedID parses a regionalNestedID and extracts the resource region, inner and outer ID.
+func parseRegionalNestedID(regionalNestedID string) (region scw.Region, outerID, innerID string, err error) {
+	locality, innerID, outerID, err := parseLocalizedNestedID(regionalNestedID)
+	if err != nil {
+		return
+	}
+
+	region, err = scw.ParseRegion(locality)
+	return
```
Comment (remyleone) [inline scaleway/helpers.go:162]: Could you be explicit in the variable you return? It makes IDE refactoring easier.

File: scaleway/resource_object.go [hunk 11]
```diff
@@ -0,0 +1,289 @@
...
+)
+
+func resourceScalewayObject() *schema.Resource {
+	return &schema.Resource{
+		CreateContext: resourceScalewayObjectCreate,
+		ReadContext:   resourceScalewayObjectRead,
+		UpdateContext: resourceScalewayObjectUpdate,
+		DeleteContext: resourceScalewayObjectDelete,
+		Timeouts: &schema.ResourceTimeout{
+			Default: schema.DefaultTimeout(defaultObjectBucketTimeout),
```
Comment (remyleone) [inline scaleway/resource_object.go:24]: Should we define all timeouts here?

File: scaleway/resource_object.go [hunk 12]
```diff
@@ -0,0 +1,289 @@
...
+					s3.ObjectCannedACLPrivate,
+					s3.ObjectCannedACLPublicRead,
+				}, false),
+			},
+			"region": regionSchema(),
+		},
+	}
+}
+
+func resourceScalewayObjectCreate(ctx context.Context, d *schema.ResourceData, meta interface{}) diag.Diagnostics {
```
Comment (remyleone) [inline scaleway/resource_object.go:85]: Should the timeout be applied in the context here?

File: scaleway/helpers.go (hunk 10 above)
Comment (Codelax) [reply scaleway/helpers.go:162]: This is a copy of `parseZonedNestedID`

**Review by remyleone — APPROVED** [review]

**Review by Monitob — APPROVED** [review]

**Pull request merged**